tests/profiles/glyf_test.py

```diff
@@ -70,3 +70,15 @@ def test_check_glyf_non_transformed_duplicate_components():
     assert_results_contain(check(ttFont),
                            FAIL, 'found-duplicates')
 
+
+def test_check_glyf_nested_components():
+    """Check glyphs do not have nested components."""
+    check = CheckTester(opentype_profile,
+                        "com.google.fonts/check/glyf_nested_components")
+
+    ttFont = TTFont(TEST_FILE("nunito/Nunito-Regular.ttf"))
+    assert_PASS(check(ttFont))
+
+    ttFont['glyf']['quotedbl'].components[0].glyphName = "second"
+    assert_results_contain(check(ttFont),
+                           FAIL, 'found-duplicates')
```

Comment:
Please add comments explaining why this test case is good for ensuring the proper behaviour of the check. Always make the code-tests more explicit about what's going on ;-)

There seem to be some hidden assumptions about the contents of the font file you're using here for testing. Please describe them so that others will be aware of it all as well.